### PR TITLE
feat: LIP-26 CSM

### PR DESCRIPTION
**Pull request by dgusakov**

### CODE REVIEW

File: LIPS/lip-26.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+## Motivation
+
+The fact that the Lido Validator set is permissioned makes it impossible for the community stakers to participate. Since community stakers participation is one of the top priorities of Ethereum, it is crucial to add a permissionless entry to the Lido on Ethereum Node Operator set and enfranchise solo-staker involvement in the protocol to stay aligned with Ethereum. At the same time, it is vital to maintain Lido on Ethereum protocol security, reliability, and capital efficiency on a level equal to or above the existing ones.
+
+Given the Staking Router architecture introduced in the [Lido V2](https://blog.lido.fi/introducing-lido-v2/) upgrade, it is proposed to implement a staking module that will add a permissionless entry to the Lido on Ethereum Node Operator set, enfranchise solo-staker participation in the protocol, and maintain Lido on Ethereum protocol security, reliability, and capital efficiency on a level equal to or above the existing ones.
```
Comment (zuzueeka) [inline LIPS/lip-26.md]: `security, reliability, and capital efficiency on a level equal to or above the existing ones.`
Amen. 

Do we really need to repeat it after each Paragraph? I mean, there are the same words at the end of the previous paragraph

File: LIPS/lip-26.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+**`CSFeeOracle.sol`** (FeeOracle on the scheme) - a utility contract responsible for the execution of the CSM Oracle report once the consensus is reached in the `HashConsensus.sol` contract, namely, transforming non-distributed rewards to non-claimed rewards stored on the `CSFeeDistributor.sol`, and reporting the latest root of rewards distribution Merkle tree to the `CSFeeDistributor.sol`.  Inherited from the `BaseOracle.sol` from LoE.
+
+**`HashConsensus.sol`** - a utility contract responsible for reaching consensus between CSM Oracle members. Uses the standard code of the `HashConsensus` contract from the LoE.
+
+**`EasyTrack`** - a utility contract responsible for the application of the reported EL stealing penalties. A part of the common `EasyTrack` setup within LoE.
```
Comment (zuzueeka) [inline LIPS/lip-26.md]: EasyTrack -> EasyTrack.sol but more likely the `particular name of the ET factory`.sol is meant here or Executor.
Or if Easy Track as a full "processor" is meant then it's not right to name it a contract.

File: LIPS/lip-26.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+
+**`GateSeal`** - a utility contract responsible for the one-time pause of the `CSModule.sol` and `CSAccounting.sol` contracts to prevent possible exploitation of the module through zero-day vulnerability. Uses the standard code of the `GateSeal` contract from LoE.
```
Comment (zuzueeka) [inline LIPS/lip-26.md]: GateSeal -> GateSeal.sol?

Comment (dgusakov) [reply LIPS/lip-26.md]: Not really, since GateSeal.sol is a factory

File: LIPS/lip-26.md (hunk 1 above)
Comment (dgusakov) [reply LIPS/lip-26.md]: Fair. Will remove the duplicate

Comment (dgusakov) [reply LIPS/lip-26.md]: Fixed

File: LIPS/lip-26.md (hunk 2 above)
Comment (dgusakov) [reply LIPS/lip-26.md]: Fixed

File: LIPS/lip-26.md (hunk 3 above)
Comment (dgusakov) [reply LIPS/lip-26.md]: NVM. Fixed

**Review by TheDZhon — APPROVED** [review]
I like this LIP :+1: 

The only concern I have is a bit of advertisement language used for Simple Summary and Motivation, I would strive to make it more neutral, less catchy, and tech-oriented if possible

File: LIPS/lip-26.md (hunk 1 above)
Comment (TheDZhon) [reply LIPS/lip-26.md]: Yeah, I believe it's a bit more for marketing activities rather than for technical specs :pray:

**Review by TheDZhon — APPROVED** [review]
Thank you for addressing the feedback smoothly!

File: LIPS/lip-26.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+
+**`CSFeeDistributor.sol`** (FeeDistributor on the scheme) - a supplementary contract that stores non-claimed and non-distributed Node Operator rewards on balance and the latest root of a rewards distribution Merkle tree root. It accepts calls from `CSAccounting.sol` with reward claim requests and stores data about already claimed rewards by the Node Operator. It receives non-distributed rewards from the `CSModule.sol` each time the `StakingRouter` mints the new portion of the Node Operators' rewards.
+
+**`CSFeeOracle.sol`** (FeeOracle on the scheme) - a utility contract responsible for the execution of the CSM Oracle report once the consensus is reached in the `HashConsensus.sol` contract, namely, transforming non-distributed rewards to non-claimed rewards stored on the `CSFeeDistributor.sol`, and reporting the latest root of rewards distribution Merkle tree to the `CSFeeDistributor.sol`.  Inherited from the `BaseOracle.sol` from LoE.
+
+**`HashConsensus.sol`** - a utility contract responsible for reaching consensus between CSM Oracle members. Uses the standard code of the `HashConsensus` contract from the LoE.
```
Comment (zuzueeka) [inline LIPS/lip-26.md]: Here and after: LoE is not commonly used abbreviation, better to avoid using it

File: LIPS/lip-26.md [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+
+**`CSMSettleELStealingPenalty.sol`** (EasyTrack on the scheme) - a utility contract responsible for the application of the reported EL stealing penalties. A part of the common `EasyTrack` setup within LoE.
+
+**`GateSeal.vy`** (GateSeal on the scheme) - a utility contract responsible for the one-time pause of the `CSModule.sol`, `CSAccounting.sol`, and `CSFeeOracle.sol` contracts to prevent possible exploitation of the module through zero-day vulnerability. Created using `GateSealFactory` contract from LoE.
```
Comment (zuzueeka) [inline LIPS/lip-26.md]: I'd better mention who is the sealing committee here

File: LIPS/lip-26.md [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+
+#### Off-chain tools
+
+**`CSM Bot`** - a daemon application responsible for monitoring and reporting the withdrawal and slashing events associated with the CSM validators.
+
+**`EL stealing detector`** - a daemon application or EOA responsible for detecting and reporting the EL stealing facts by the CSM validators. Assumed to be EOA controlled by the dev team at the early stages of the MEV monitoring software maturity and later converted to the automated bot to avoid false-positive activations.
+
+**`CSM Oracle`** - a module in the common LoE oracle set. Operated by the existing oracles set alongside Accounting Oracle and Validator Exit Bus Oracle. It is responsible for the calculation of the CSM Node Operators' rewards distribution based on their attestation performance on the CL.
```
Comment (zuzueeka) [inline LIPS/lip-26.md]: what does it mean "a module in the common LoE oracle set."?

File: LIPS/lip-26.md (hunk 4 above)
Comment (dgusakov) [reply LIPS/lip-26.md]: Replaced with Lido on Ethereum

File: LIPS/lip-26.md (hunk 5 above)
Comment (dgusakov) [reply LIPS/lip-26.md]: Note added

File: LIPS/lip-26.md (hunk 6 above)
Comment (dgusakov) [reply LIPS/lip-26.md]: Clarifications and links were added

**Review by zuzueeka — APPROVED** [review]
I can't fully grasp the content in its entirety at first glance, as it's quite challenging for me. However, from a logical and design perspective, I give it a thumbs up.

**Pull request merged**